inc/ad-servers/class-ad-layers-dfp.php

```diff
@@ -0,0 +1,1176 @@
+		/**
+		 * Available ad units on the page.
+		 *
+		 * @access public
+		 * @var array
+		 */
+		public $ad_units = [];
+
+		/**
+		 * Size mappings by unit.
+		 *
+		 * @access public
+		 * @var array
+		 */
+		public $mapping_by_unit = [];
+
+		/**
+		 * Default sizes by unit.
+		 *
+		 * @access public
+		 * @var array
+		 */
+		public $default_by_unit = [];
+
+		/**
+		 * Raw values for targeting by unit.
+		 *
+		 * @access public
+		 * @var array
+		 */
+		public $raw_targeting_by_unit = [];
+
+		/**
+		 * Targeting by unit.
+		 *
+		 * @access public
+		 * @var array
+		 */
+		public $targeting_by_unit = [];
+
+		/**
+		 * Out of page units.
+		 *
+		 * @access public
+		 * @var array
+		 */
+		public $oop_units = [];
+
+		/**
+		 * Cache key
+		 *
+		 * @access public
+		 * @var string
+		 */
+		public $cache_key = 'ad_layers_dfp_settings';
+
+		/**
+		 * Javascript API class name
+		 *
+		 * @access public
+		 * @var string
+		 */
+		public $js_api_class = 'AdLayersDFPAPI';
+
+		/**
+		 * Handle used for scripts
+		 *
+		 * @access public
+		 * @var string
+		 */
+		public $handle = 'ad-layers-dfp';
+
+		/**
+		 * List of ads for which to skip rendering.
+		 *
+		 * @var array
+		 */
+		public $do_not_render_ads = [];
+
+		/**
+		 * List of ads for which to skip rendering.
+		 *
+		 * @var array
+		 */
+		public $ad_unit_paths = [];
+
+		/**
+		 * Setup the singleton.
+		 */
+		public function setup() {
+			// Define the available formatting tags.
+			$this->formatting_tags = apply_filters( 'ad_layers_dfp_formatting_tags', $this->set_formatting_tags() );
+
+			// Allow filtering of the ad unit prefix.
+			$this->ad_unit_prefix = apply_filters( 'ad_layers_dfp_ad_unit_prefix', $this->ad_unit_prefix );
+
+			// Add a help tab.
+			add_action( 'load-' . Ad_Layers_Post_Type::instance()->get_post_type() . '_page_' . $this->option_name, [ $this, 'add_help_tab' ] );
+
+			// Handle caching.
+			add_action( 'update_option', [ $this, 'cache_settings' ], 10, 3 );
+
+			// Add the path override field to the ad layer ad units.
+			add_filter( 'ad_layers_ad_units_field_args', [ $this, 'ad_layer_path_overrides' ] );
+		}
+
+		/**
+		 * Load scripts.
+		 *
+		 * @access public
+		 */
+		public function enqueue_scripts() {
+			// Load the base Javascript library (in header to ensure early ad loading).
+			wp_enqueue_script(
+				$this->handle,
+				get_asset_path( 'adLayersDfp.js' ),
+				[ 'jquery' ],
+				get_asset_hash( 'adLayersDfp.js' ),
+				false
+			);
+
+			// Localize the base API with static text strings so they can be translated.
+			wp_localize_script(
+				$this->handle,
+				'adLayersDFP',
+				[
+					'layerDebugLabel'   => __( 'Current ad layer', 'ad-layers' ),
+					'consoleDebugLabel' => __( 'Switch to Google console', 'ad-layers' ),
+					'adUnitPrefix'      => $this->ad_unit_prefix,
+				]
+			);
+		}
+
+		/**
+		 * Set the available formatting tags.
+		 *
+		 * Should be implemented by all child classes.
+		 *
+		 * @access public
```

Comment:
I usually find those `@access` tags unnecessary since one can see quickly they are public/private/protected. They are easy to search for in an IDE. I'm not sure how helpful they actually are.